collector/mdadm.go

```diff
@@ -0,0 +1,200 @@
+// +build !nomdadm
+
+package collector
+
+import (
+	"io/ioutil"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/log"
+)
+
+var statusfile = "/proc/mdstat"
+
+type mdstatus struct {
+	mdName       string
+	isActive     bool
+	disksActive  int64
+	disksTotal   int64
+	blocksTotal  int64
+	blocksSynced int64
+}
+
+type mdadmCollector struct {
+	isActive     *prometheus.GaugeVec
+	disksActive  *prometheus.GaugeVec
+	disksTotal   *prometheus.GaugeVec
+	blocksTotal  *prometheus.GaugeVec
+	blocksSynced *prometheus.GaugeVec
+}
+
+func init() {
+	Factories["mdadm"] = NewMdadmCollector
+}
+
+func evalStatusline(statusline string) (active, total, size int64) {
```

Comment:
I saw the discussion on IRC about using regexes. I think this would be perfectly fine here, and probably more understandable. There are already other collectors also using regexes, like netdev and meminfo.